File: School/CSUEB/Fall 2013/CS-3120/Proj5/UserInput C++/UserInput.cpp
//
//  UserInput.cpp
//  Gamegame
//
//  Created by Nicholas Rebhun on 11/12/13.
//  Copyright (c) 2013 Nicholas Rebhun. All rights reserved.
//

#include "UserInput.h"

string UserInput::GetString(){
	string theStr;
	cout << "> ";
	getline(cin, theStr);
	return theStr;
}

int UserInput::GetInt() {
	int i;
	string theIntStr = GetString();
	stringstream convert(theIntStr);
	convert >> i;
	return i;
}

char UserInput::GetChar() {
	string tempStr = GetString();
	return tempStr[0];
}

double UserInput::GetDouble() {
	string doubStr = GetString();
	return atof(doubStr.c_str()	);
}

bool UserInput::GetConfirmation() {
	cout << " (y/n): ";
	char theChar = GetChar();
	while (theChar != 'y' && theChar != 'n') {
		cout << "Uh-oh! Input must be either 'y' or 'n'.\nTry Again." << endl;
		theChar = GetChar();
	}
	if (theChar == 'y') {
		return 1;
	}
	else if (theChar == 'n') {
		return 0;
	}
}

string UserInput::GetString(int min, int max) {
	string theString = GetString();
	while (theString.length() < min || theString.length() > max) {
		cout << "Uh-oh! You entered: " << theString << ", which is " << theString.length() << "characters in length.\nInput must be between " << min << " and " << max << " characters.\nTry again." << endl;
		theString = GetString();
	}
	return theString;
}

int UserInput::GetInt(int min, int max) {
	int theInt = GetInt();
	while ((theInt <= min) && (theInt >= max)) {
		cout << "Uh-oh! Input must be between " << min << " and " << max << ".\nTry again." << endl;
		theInt = GetInt();
	}
	return theInt;
}

char UserInput::GetChar(char min, char max) {
	char theChar = GetChar();
	while (theChar < min || theChar > max) {
		cout << "Uh-oh! Input must be between " << min << " and " << max << ".\nTry again." << endl;
		theChar = GetChar();
	}
	return theChar;
}

File: School/Misc CS/Java/182/Programs/Project 1/src/project/pkg1/UserInput.java
package project.pkg1;

import java.util.*;

public class UserInput {

    public static String getString() {
        Scanner isr = new Scanner(System.in);
        String s = isr.nextLine();
        return s;
    }

    public static char getChar() {
        String s = getString();
        return s.charAt(0);
      }

    public static int getInt() {
        String s = getString();
        return Integer.parseInt(s);
    }

    public static double getDouble() {
        String s = getString();
        Double aDub = Double.valueOf(s);
        return aDub.doubleValue();
    }

    public static int getInt(int min, int max) {
        int s = getInt();

        while(s < min || s > max) {
            System.out.println("That input is not valid.\nTry again:");
            s = getInt();
        }
        return s;
    }

    public static char getChar(char min, char max) {
        char s = getChar();
        while(s < min || s > max) {
            System.out.println("That input is not valid.\nTry again:");
            s = getChar();
        }
        return s;
    }

    public static double getDouble(double min, double max) {
        double s = getDouble();
        while(s < min || s > max) {
            System.out.println("That input is not valid.\nTry again:");
            s = getDouble();
        }
        return s;
    }

    public static String getString(int min, int max) {
        String s = getString();
        while(s.length() < min || s.length() > max) {
            System.out.println("That input is not valid.\nTry again:");
            s = getString();
        }
        return s;
    }
}